### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by layzrd**
### In this milestone, I

- Added a weather display page for multiple cities
- Added search feature for searching by city
- Added error handling
- Added notification feature
- Configured Webpack to build process
- Configured babel to support old browsers

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -1,150 +1,145 @@
...
+      },
+      redirect: 'follow',
+      referrerPolicy: 'no-referrer',
+      body: JSON.stringify(data),
+    }).then((response) => {
+      if (response.ok) {
+        return response.json();
+      }
+      return Promise.reject(response);
+    }).catch(handleError);
```
Comment (lucasmazo23) [inline src/api.js]: - [ ] [REQUIRED] Here you are mixing concepts, if you are already using await, you don't need the `.then` at the end. Also if you use `async/await` you need to wrap your login inside of a `try/catch` statement to handle errors
image
[Source](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/)

**Review by lucasmazo23 — CHANGES_REQUESTED** [review]
## Changes required 🔧
### You are doing a great job so far, and the project is looking good.
 
### For the project to get approved kindly implement these changes:
 
### Required changes:
- [ ] [REQUIRED] I couldn't open the app, both dev server and live version give me errors on the chrome console
  - live error: 
image
  - dev error: 
image

- [ ] [REQUIRED] Check my comments below :D

### Happy coding! 💻
## Lucas Mazo

File: src/api.js (hunk 1 above)
Comment (addod19) [reply src/api.js]: This comment is not addressed. Kindly address it

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @abredi, great job on this project :+1: However, we did not address the comments from the previous TSE :slightly_smiling_face: Let us kindly address the previous comments

- Webpack is not configured correctly. Running `npm run dev` outputs the error below. It is probably a loader error
asc
Kindly fix it
- The live version has no button to toggle between Celsius and Fahrenheit. Let us kindly fix it
-  Async/await goes with `try/catch` and promises goes with `then/catch`

Once these are fixed, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Comment by layzrd** [comment]
Hi @addod19, thank you for your review, I fixed the `url-loader` issues it was a glitch between yarn and npm. But, the toggler is already there on the page header and I mixed up `async` and `then` for housekeeping (to catch unpredictable fetching errors) and readability purpose.

[Abdulaziz Redi Ali](https://azizali.ml/) | [Contact me on Slack](https://microverse-students.slack.com/team/U011SD6JUH2)

File: src/api.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import BASE_URL from './constants';
+import { clearContent, notification } from './util.dom';
+
+import KEY from '../key'; // eslint-disable-line
```
Comment (trekab) [inline src/api.js:4]: I am unable to run your app locally with `npm run dev` because the `key` module is missing.

**Review by trekab — CHANGES_REQUESTED** [review]
### Changes Required!
Hi @abredi! It appears that you have handled the required changes but I still get an error when I try to run your app locally. I have commented on the source of the error below in the code. Please fix this.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[Portfolio](https://trekab.github.io/portfolio/) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

File: src/api.js (hunk 2 above)
Comment (layzrd) [reply src/api.js:4]: Hi @trekab, thanks for your review. Its explained on the readme. Kindly check the local setup instructions in the readme file.

Comment (trekab) [reply src/api.js:4]: I missed it, sorry about that. Please resubmit this code review request.

**Review by sumancrest0001 — APPROVED** [review]
**Status: Approved:100: :+1: 
Well done @abredi :clap:** 
It is approved as the weather app is working as expected. Design and file structure looks good to me. Now, You can merge the branch to master. Before that, I encourage you to have a look at the following OPTIONAL changes which are good features to add.
1. It has a ‘loading’ component that displays from the time the form is submitted until the information comes back from the API. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/loading_component_requirement.png)
2. It fetches the current geolocation of the user when the app is first accessed and fetches the weather from that location

**Pull request merged**